src/Input/RealInput.st

```diff
@@ -39,5 +60,15 @@ NAMESPACE Simatic.Ax.IO.Input
             END_IF;     
         END_METHOD
 
+        /// Returns TRUE if current value is not equal to last value
+        /// within a tolerance
+        METHOD PUBLIC HasChanged : BOOL
+            VAR_INPUT
+                Tolerance : REAL := REAL#0.0;
+            END_VAR
+
+            HasChanged := ABS(_value - _last_value) > Tolerance;
```

Comment:
Idea: implement the HasChanged-Method in the InputBase-Class. Then you need just to implement to assign the value in the ReadCyclic part.

METHOD HasChanged : BOOL
   HasChanged := _hasChanged;
END_METHOD


METHOD ReadCyclic
	_hasChanged := _value <> _oldValue;
END_METHOD

Reply:
I will change this to be implemented through InputBase instead, the tolerance will be fed through ReadCyclic  (for all scalars/analogs) and the comparison done there.

Reply:
thanks